algoliasearch/index_iterator.go

```diff
@@ -15,21 +13,17 @@ type IndexIterator struct {
 // newIndexIterator instantiates a IndexIterator on the `index` and according
 // to the given `params`. It is also trying to load the first page of results
 // and return an error if something goes wrong.
-func newIndexIterator(index *Index, params Map) (it IndexIterator, err error) {
-	it = IndexIterator{
+func newIndexIterator(index Index, params Map) (it IndexIterator, err error) {
+	it = &indexIterator{
 		index:  index,
 		params: duplicateMap(params),
 		pos:    0,
 	}
-	err = it.loadNextPage()
+	err = it.(*indexIterator).loadNextPage()
```

Comment:
Agree.

However, are you sure it's useful to have an `interface` for the `indexIterator`? In the end, it's just an implementation detail of the `index` structure. I think we should keep the `IndexIterator` as it was previously (not an interface). Final users who would like to mock the `index` will probably change all the `Browse*` functions according to their needs, without reusing the `indexIterator`. @ercolanelli-leo What do you think?